File: go-version/leetcode/LC2598/find_smallest_integer.go
package LC2598

/**
 * @Author: cutejiuge cutejiuge@163.com
 * @Date: 2025/10/16 下午11:38
 * @FilePath: find_smallest_integer
 * @Description: 同余计数
 */

func findSmallestInteger(nums []int, value int) int {
	mp := make([]int, value)
	for _, num := range nums {
		v := (num%value + value) % value
		mp[v]++
	}

	var mex int
	for mp[mex%value] > 0 {
		mp[mex%value]--
		mex++
	}
	return mex
}

File: python-version/LC2598/find_smallest_integer.py
from typing import List


class Solution:
    @staticmethod
    def findSmallestInteger(nums: List[int], value: int) -> int:
        mp = [0] * value
        for num in nums:
            v = (num % value + value) % value
            mp[v] += 1

        mex = 0
        while mp[mex % value] > 0:
            mp[mex % value] -= 1
            mex += 1
        return mex
